src/index.js

```diff
@@ -93,17 +92,34 @@ class Root extends React.Component {
   }
 }
 
-store.dispatch(fetchConfig());
-store.dispatch(fetchFilterConfig());
+const loadConfig = (route) => {
+  return fetch(route).then(res => res.json());
+};
+
+// entry point for the app
+Promise.all([loadConfig('/config.json'), loadConfig('/data_config.json')])
+  .then(results => {
+
+    const [app, filters] = results;
+
+    const requiredKeys = [ 'xeniaHost', 'pillarHost', 'basicAuthorization', 'environment', 'googleAnalyticsId', 'requireLogin' ];
+    const allKeysDefined = requiredKeys.every(key => 'undefined' !== typeof app[key]);
+
+    if (!allKeysDefined) {
+      throw new Error(`missing required keys on config.json. Must define ${requiredKeys.join('|')}`);
+    }
+
+    // load config into initialState so it's ALWAYS available
+    store = configureStore({app});
```

Comment:
This is an interesting pattern - haven't seen it before but seems ok. I do think it's ok to initialize redux before this occurs - in that case app has an example to show a spinner and error messaging for the app based on what's in the store since it's initialized.